pkg/workflow/manager.go

```diff
@@ -299,38 +339,139 @@ func (w *WorkflowManager) enqueueController(obj interface{}) {
 }
 
 func (w *WorkflowManager) addJob(obj interface{}) {
+	job := obj.(*k8sBatch.Job)
+	glog.V(3).Infof("addJob %v", job.Name)
+	if workflow := w.getJobWorkflow(job); workflow != nil {
+		w.enqueueController(workflow)
+	}
 }
 
 func (w *WorkflowManager) updateJob(old, cur interface{}) {
-
+	oldJob := old.(*k8sBatch.Job)
+	curJob := cur.(*k8sBatch.Job)
+	glog.V(3).Infof("updateJob old=%v, cur=%v ", oldJob.Name, curJob.Name)
+	if k8sApi.Semantic.DeepEqual(old, cur) {
+		glog.V(3).Infof("\t nothing to update")
+		return
+	}
+	if workflow := w.getJobWorkflow(curJob); workflow != nil {
+		glog.V(3).Infof("enqueueing controller for %v", curJob.Name)
+		w.enqueueController(workflow)
+	}
 }
 
 func (w *WorkflowManager) deleteJob(obj interface{}) {
-
+	job, ok := obj.(*k8sBatch.Job)
+	if !ok {
+		tombstone, ok := obj.(k8sCache.DeletedFinalStateUnknown)
+		glog.V(3).Infof("Tombstone found %v", tombstone)
+		if !ok {
+			glog.Errorf("Couldn't get object from tombstone %+v", obj)
```

Comment:
This seems an odd error message, if the type assertion on 366 fails it means that the object is not a tombstone but is in fact something else, right?